pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -467,9 +468,24 @@ func (d *Driver) sendSignal(s syscall.Signal) error {
 }
 
 func (d *Driver) supportsVirtiofs() bool {
-	macosVersion, _, err := crcos.RunWithDefaultLocale("sw_vers", "-productVersion")
+	supportsVirtioFS, err := macosNewerThan(11.0)
 	if err != nil {
-		return false
+		log.Debugf("Not able to compare version: %v", err)
 	}
-	return strings.HasPrefix(macosVersion, "12.")
+	return supportsVirtioFS
+}
+
+func macosNewerThan(targetVersion float64) (bool, error) {
+	osVer, err := syscall.Sysctl("kern.osproductversion")
+	if err != nil {
+		return false, err
+	}
+	logging.Debugf("Mac version is: %s", osVer)
+	osVerArray := strings.SplitAfterN(osVer, ".", 1)
+	logging.Debugf("Mac major version is: %s", osVerArray)
+	version, err := strconv.ParseFloat(osVerArray[0], 64)
```

Comment:
This fails to parse `12.6.1` (and most likely `13.0.1` will have a similar failure). I'm unconvinced a float is a good way to represent the macOS version, I'd either go with an int for just the major version, or with `semver.NewVersion()` if we want the full version.

Reply:
Agree. A float sounds awkward